examples_tests/51.RadixSort/main.cpp

```diff
@@ -0,0 +1,287 @@
+
+	auto downloadFence = driver->placeFence(true);
+	auto result = downloadFence->waitCPU(timeout_ns, true);
+
+	T* dataFromBuffer = nullptr;
+	if (result != video::E_DRIVER_FENCE_RETVAL::EDFR_TIMEOUT_EXPIRED && result != video::E_DRIVER_FENCE_RETVAL::EDFR_FAIL)
+	{
+		if (downloadStagingArea->needsManualFlushOrInvalidate())
+			driver->invalidateMappedMemoryRanges({ {downloadStagingArea->getBuffer()->getBoundMemory(),address,array_size_32} });
+
+		dataFromBuffer = reinterpret_cast<T*>(reinterpret_cast<uint8_t*>(downloadStagingArea->getBufferPointer()) + address);
+	}
+	else
+	{
+		os::Printer::log("Could not download the buffer from the GPU, fence not signalled!", ELL_ERROR);
+	}
+
+	downloadStagingArea->multi_free(1u, &address, &array_size_32, nullptr);
+
+	return dataFromBuffer;
+}
+
+template <typename T>
+static void DebugCompareGPUvsCPU(smart_refctd_ptr<IGPUBuffer> gpu_buffer, T* cpu_buffer, size_t buffer_size, IVideoDriver* driver)
+{
+	T* downloaded_buffer = DebugGPUBufferDownload<T>(gpu_buffer, buffer_size, driver);
+
+	size_t buffer_count = buffer_size / sizeof(T);
+
+	if (downloaded_buffer)
+	{
+		for (int i = 0; i < buffer_count; ++i)
+		{
+			if (downloaded_buffer[i] != cpu_buffer[i])
+				__debugbreak();
+		}
+
+		std::cout << "PASS" << std::endl;
+	}
+}
+
+static void RadixSort(IVideoDriver* driver, const SBufferRange<IGPUBuffer>& in_gpu_range,
+	IGPUDescriptorSet* ds_histogram, IGPUComputePipeline* histogram_pipeline,
+	IGPUDescriptorSet*  ds_scan, IGPUComputePipeline*  upsweep_pipeline, IGPUComputePipeline* downsweep_pipeline,
+	IGPUDescriptorSet* ds_scatter, IGPUComputePipeline* scatter_pipeline)
+{
+	const uint32_t total_scan_pass_count = RadixSortClass::buildParameters(in_gpu_range.size / sizeof(SortElement), WG_SIZE,
+		nullptr, nullptr, nullptr, nullptr);
+
+	RadixSortClass::Parameters_t sort_push_constants[RadixSortClass::PASS_COUNT];
+	RadixSortClass::DispatchInfo_t sort_dispatch_info;
+
+	const uint32_t upsweep_pass_count = (total_scan_pass_count / 2) + 1;
+	core::vector<ScanClass::Parameters_t> scan_push_constants(upsweep_pass_count);
+	core::vector<ScanClass::DispatchInfo_t> scan_dispatch_info(upsweep_pass_count);
+
+	RadixSortClass::buildParameters(in_gpu_range.size / sizeof(SortElement), WG_SIZE, sort_push_constants, &sort_dispatch_info,
+		scan_push_constants.data(), scan_dispatch_info.data());
+
+	SBufferRange<IGPUBuffer> scratch_gpu_range = { 0 };
+	scratch_gpu_range.size = in_gpu_range.size;
+	scratch_gpu_range.buffer = driver->createDeviceLocalGPUBufferOnDedMem(in_gpu_range.size);
+
+	const uint32_t histogram_count = sort_dispatch_info.wg_count[0] * RadixSortClass::BUCKETS_COUNT;
+	SBufferRange<IGPUBuffer> histogram_gpu_range = { 0 };
+	histogram_gpu_range.size = histogram_count * sizeof(uint32_t);
+	histogram_gpu_range.buffer = driver->createDeviceLocalGPUBufferOnDedMem(histogram_gpu_range.size);
+
+	ScanClass::updateDescriptorSet(ds_scan, histogram_gpu_range, driver);
```

Comment:
In the spirit of https://github.com/Devsh-Graphics-Programming/Nabla/pull/75#discussion_r608107964,
I could wrap this in `RadixSort::updateDescriptorSet` overload.

Also, ScanClass is just `ext::RadixSort::ScanClass`

Reply:
good idea, do it